lib/mysqlDaoQueryHelper.js

```diff
@@ -58,6 +58,23 @@ class MysqlDaoQueryHelper {
     }
 
     /**
+     * @param {object} upsertArgs
+     * @returns {string}
+     */
+    upsert(upsertArgs) {
```

Comment:
Suggestion to throw out there: split the insert and on duplicate key update arguments.  Function would therefore take two params, insertArgs and updateOnDupKeyArgs with updateOnDupKeyArgs defaulting to insertArgs.

Should be able to support the following:
1) Update fewer / different columns than are included in insertArgs if there's a duplicate key
2) Do functional operations on column updates (e.g., increment a column that counts the number of upsert operations on a row).  This might require some finangling of how values are escaped.  Will likely need to use an object so we don't escape the entire string.

Might look something like this:

```
const insertArgs = {
  email: 'john.doe@gmail.com',
  name: 'John Doe',
  num_updates: 0
}

const updateOnDupKeyArgs = {
  name: 'John Doe',
  num_updates: {raw: 'num_updates + 1'}
}

dao.upsert(insertArgs, updateOnDupKeyArgs)
```